### PR TITLE
Fixes a bug involving plasma always being set to 0 when less than 5

### PR DESCRIPTION
**Pull request by ma44**
## About The Pull Request

Mistake with some code involving the optimization of pheromones led to this bad bug
## Why It's Good For The Game
Makes xenos with less than 5 plasma and not enough plasma regen no longer be stuck at 0 plasma

## Changelog
:cl:
fix: Xenos with low plasma regen like runners will no longer be stuck at 0 plasma when running out of it
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/life.dm [hunk 1]
```diff
@@ -141,22 +141,19 @@
 		return
 	if(current_aura)
 		plasma_stored -= 5
-	if(plasma_stored == xeno_caste.plasma_max)
+	if(plasma_stored >= xeno_caste.plasma_max)
 		return
```
Comment (psykzz) [inline code/modules/mob/living/carbon/xenomorph/life.dm]:
```suggestion
		plasma_stored = xeno_caste.plasma_max
		return
```

You should ensure they don't stay over their limits.

File: code/modules/mob/living/carbon/xenomorph/life.dm [hunk 2]
```diff
@@ -139,24 +139,24 @@
...
 		plasma_stored++
-	if(plasma_stored > xeno_caste.plasma_max)
-		plasma_stored = xeno_caste.plasma_max
-		return //Ditto above == max
-	else if(plasma_stored < 5)
+
+	if(current_aura)
+		plasma_stored -= 5
+
+	if(current_aura && plasma_stored < 5)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/life.dm]:
```suggestion
	if(current_aura && plasma_stored < 10)
```
And remove the prior check.

**Comment by DominikPanic** [comment]
Travis is reeeing.

File: code/modules/mob/living/carbon/xenomorph/life.dm [hunk 3]
```diff
@@ -139,24 +139,20 @@
...
 	else
 		plasma_stored++
+
+	if(current_aura && (plasma_stored -= 5) < 5)
+		plasma_stored = 0
+        current_aura = null
+        to_chat(src, "<span class='warning'>You have run out of plasma and stopped emitting pheromones.</span>")
+
 	if(plasma_stored > xeno_caste.plasma_max)
 		plasma_stored = xeno_caste.plasma_max
```
Comment (Jalleo) [inline code/modules/mob/living/carbon/xenomorph/life.dm]: Needs a return after this. We dont need to go to hud_set_plasma if it is already at or above max.
It is unlikely apart from a edgecase of specific numbers that its needed to update.

File: code/modules/mob/living/carbon/xenomorph/life.dm [hunk 4]
```diff
@@ -139,24 +139,20 @@
...
-	if(plasma_stored == xeno_caste.plasma_max)
-		return
 	if(locate(/obj/effect/alien/weeds) in T)
 		plasma_stored += xeno_caste.plasma_gain
 		if(recovery_aura)
 			plasma_stored += round(xeno_caste.plasma_gain * recovery_aura * 0.25) //Divided by four because it gets massive fast. 1 is equivalent to weed regen! Only the strongest pheromones should bypass weeds
 	else
 		plasma_stored++
+
+	if(current_aura && (plasma_stored -= 5) < 5)
```
Comment (Jalleo) [inline code/modules/mob/living/carbon/xenomorph/life.dm]: Is this removing five plasma inside the check? I cant remember if it does or not. I am not entirely remembering correctly.

**Comment by psykzz** [comment]
Can you just replace the entire proc with 
```
/mob/living/carbon/xenomorph/proc/handle_living_plasma_updates()
	var/turf/T = loc
	if(!T || !istype(T))
		return
	if(plasma_stored == xeno_caste.plasma_max)
		return

	if(current_aura)
		if(plasma_stored < 5)
			use_plasma(plasma_stored)
			current_aura = null
			to_chat(src, "<span class='warning'>You have run out of plasma and stopped emitting pheromones.</span>")
		else
			use_plasma(5)

	if(locate(/obj/effect/alien/weeds) in T)
		gain_plasma(xeno_caste.plasma_gain + round(xeno_caste.plasma_gain * recovery_aura * 0.25)) // Empty recovery aura will always equal 0
	else
		gain_plasma(1)

	hud_set_plasma() //update plasma amount on the plasma mob_hud
```

**Pull request merged**